# Importation des bibliothèques

In [19]:
import pandas as pd

# Constantes

In [20]:
# chemin pour accéder aux données
FILE_PATH = '../src/data/donnees_immobilieres.parquet'

# Application

| Nom de colonne                 | Description |
|--------------------------------|-------------|
| region                         | Région où se situe le bien |
| ue                             | Identifiant de l’Unité économique (Site). À compter du 31/10/2015, remplacé par l’ID unique du bien |
| id                             | Identifiant unique du bien |
| type                           | Usage majoritaire du bien |
| fonction                       | Usage majoritaire détaillé du bien |
| designation_batiment_terrain   | Désignation du bâtiment ou terrain |
| dept                           | Code du département de situation du bien |
| adresse                        | Adresse normalisée du bien |
| code_postal                    | Code postal de situation du bien |
| ville                          | Ville de situation du bien |
| pays                           | Pays de situation du bien |
| ministere                      | Ministère occupant le bien |
| date_inventaire                | Date de l’inventaire |

In [21]:
# Chargement du DataFrame depuis le fichier Parquet 
df = pd.read_parquet(FILE_PATH)

In [22]:
df.head()

,region,ue,id,type,fonction,designation_batiment_terrain,dept,adresse,code_postal,ville,pays,ministere,date_inventaire
0,CORSE,a19e7cbea4c2dae3cfca703bda49e466,936465ba5b332c60491f6cba50a76b49,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,2B,None,None,None,France,"Intérieur, Outre-mer et Collec. terr.",2014-03
1,NORD-PAS DE CALAIS,c1fb47969da8dd1a5a1a93ae33245ced,44b7ec415ee6f810d44fda9c8c585069,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,62,None,None,None,France,Intérieur - Outre-mer - Collectivités territor...,2015-10
2,ILE DE France,120994,IG00100036786,Espace naturel,Terrain en friche,MIN de Rungis,94,LD LA BUTTE DE CHEVILLY,94260,Fresnes,France,Services du Premier ministre,2014-03
3,RHONE ALPES,152769,IB00100093449,Bâtiment technique,Bâtiment technique,GENDARMERIE DE SAINT GENIS LAVAL,69,92 Avenue MAL FOCH,69000,SAINT-GENIS-LAVAL,France,France Domaine,2013-05
4,BASSE NORMANDIE,126681,IB00100205180,Bâtiment technique,Dépôt d'archives,ARCHIVES JUSTICE CALVADOS (pré-archivage judic...,14,R NICEPHORE NIEPCE,14120,Mondeville,France,Justice,2017-12


In [31]:
# Dimensions du DataFrame 
print(f"Nombre de lignes : {df.shape[0]}")
print(f"Nombre de colonnes : {df.shape[1]}")

Nombre de lignes : 10000
Nombre de colonnes : 13


In [23]:
# Informations générales sur le DataFrame 
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 13 columns):
 #   Column                        Non-Null Count  Dtype 
---  ------                        --------------  ----- 
 0   region                        10000 non-null  object
 1   ue                            9703 non-null   object
 2   id                            10000 non-null  object
 3   type                          10000 non-null  object
 4   fonction                      10000 non-null  object
 5   designation_batiment_terrain  10000 non-null  object
 6   dept                          9998 non-null   object
 7   adresse                       6712 non-null   object
 8   code_postal                   6985 non-null   object
 9   ville                         6972 non-null   object
 10  pays                          10000 non-null  object
 11  ministere                     10000 non-null  object
 12  date_inventaire               10000 non-null  object
dtypes: object(13)
mem

In [50]:
# Statistiques descriptives pour les données numériques
df.describe()

,region,ue,id,type,fonction,designation_batiment_terrain,dept,adresse,code_postal,ville,pays,ministere,date_inventaire
count,10000,9703,10000,10000,10000,10000,9998,6712,6985,6972,10000,10000,10000
unique,25,5110,9041,18,107,5515,110,5697,2310,3775,91,46,6
top,ILE DE France,110056,IG00100014766,Espace naturel,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,13,LD LE BOURG,85460,La Faute-sur-Mer,France,Ecologie - Développement durable - Energie - A...,2013-05
freq,1171,54,4,3887,2989,2989,326,20,127,54,9823,2518,1881


# 🏢 Interprétation des statistiques descriptives (df.describe)

## 1. Vue d’ensemble
Le dataset contient **10 000 biens immobiliers publics** avec 13 colonnes :
- Identifiants (`ue`, `id`)
- Localisation (`region`, `dept`, `adresse`, `code_postal`, `ville`, `pays`)
- Usage (`type`, `fonction`, `designation_batiment_terrain`)
- Ministère occupant (`ministere`)
- Date de l’inventaire (`date_inventaire`)

---

## 2. Analyse par colonnes

### Identifiants
- **id** : 10 000 valeurs mais seulement 9 041 uniques → certains identifiants apparaissent plusieurs fois (doublons à traiter).
- **ue** : 9 703 valeurs renseignées, 5 110 uniques. Valeurs manquantes (≈3 %) car remplacé par l’`id` après 2015.

### Localisation
- **region** : 25 valeurs distinctes, cohérent avec les anciennes régions françaises. Valeur la plus fréquente : *Île-de-France* (1171 biens).
- **dept** : 110 départements distincts. 2 valeurs manquantes.
- **ville** : 6 972 valeurs renseignées (≈70 %). 3 775 villes distinctes. Valeur la plus fréquente : *La Faute-sur-Mer* (54 occurrences).
- **code_postal** : 6 985 valeurs renseignées (≈70 %), 2 310 distincts.
- **adresse** : seulement 6 712 adresses renseignées (≈67 %), 5 697 distinctes.
- **pays** : quasi exclusivement *France* (98,2 %).

### Usage
- **type** : 18 catégories différentes. Valeur la plus fréquente : *Espace naturel* (3 887 biens).
- **fonction** : 107 usages détaillés. Valeur la plus fréquente : *biens du ministère de l’Intérieur* (2 989 occurrences).
- **designation_batiment_terrain** : 5 515 désignations distinctes, mais redondance forte avec *biens du ministère de l’Intérieur* (2 989 occurrences).

### Ministère
- **ministere** : 46 ministères distincts. Le plus représenté : *Écologie - Développement durable - Énergie ...* (2 518 biens).

### Date d’inventaire
- **date_inventaire** : seulement 6 valeurs distinctes (ex. 2013-05, 2014-03, 2015-10…). Cela correspond à des campagnes d’inventaire ponctuelles.

---

## 3. Points clés pour le nettoyage
- 🔍 **Doublons** : vérifier `id`, qui devrait être unique.
- ⚠️ **Valeurs manquantes** : fortes proportions sur `adresse`, `code_postal`, `ville`.
- 🧹 **Normalisation** :
  - Harmoniser la casse (`ville`, `region`, `ministere`).
  - Nettoyer les désignations redondantes.
- 📊 **Colonnes peu informatives** :
  - `pays` (quasi toujours *France*).
  - `date_inventaire` (peu de valeurs distinctes).

In [51]:
df["region"].value_counts(normalize=True) * 100

region
ILE DE France           11.71
D.O.M                    8.28
PACA                     7.54
BRETAGNE                 7.50
RHONE ALPES              6.63
MIDI-PYRENNEES           5.42
AQUITAINE                5.15
PAYS DE LOIRE            5.06
NORD-PAS DE CALAIS       4.81
POITOU                   4.78
LORRAINE                 3.87
CENTRE                   3.43
LANGUEDOC-ROUSSILLON     3.42
PICARDIE                 2.58
AUVERGNE                 2.47
CHAMPAGNE-ARDENNES       2.28
HAUTE NORMANDIE          2.16
BOURGOGNE                2.14
FRANCHE COMTE            2.13
BASSE NORMANDIE          2.02
ALSACE                   1.93
ETRANGER                 1.77
LIMOUSIN                 1.51
CORSE                    1.06
C.O.M                    0.35
Name: proportion, dtype: float64

In [38]:
df["type"].value_counts(normalize=True) * 100

type
Espace naturel                          38.87
biens du ministère de l'Intérieur       29.89
Bureau                                  11.18
Bâtiment technique                       8.25
Bâtiment sanitaire ou social             2.39
Ouvrage d'art des réseaux et voiries     1.77
Bât. enseignement ou sport               1.42
Espace aménagé                           1.40
Bâtiment d'enseignement ou de sport      1.17
Bâtiment culturel                        0.82
Ouvrages d'art réseaux voiries           0.78
Réseaux et voirie                        0.66
Réseaux et voiries                       0.38
Monument et mémorial                     0.28
Bâtiment agricole ou d'élevage           0.27
Édifice du culte                         0.23
Edifice du culte                         0.15
Commerce                                 0.09
Name: proportion, dtype: float64

In [39]:
df["fonction"].value_counts(normalize=True) * 100

fonction
biens du ministère de l'Intérieur    29.89
Terrain en friche                    19.56
Terrain délaissé                     13.31
Immeuble de bureaux                   7.10
Bâtiment technique                    2.40
                                     ...  
Salle de spectacles                   0.01
Dune                                  0.01
Ne plus utiliser                      0.01
Crèche                                0.01
Bâtiment de logements meublés         0.01
Name: proportion, Length: 107, dtype: float64

In [40]:
df["ville"].value_counts(normalize=True) * 100

ville
La Faute-sur-Mer            0.774527
L'Aiguillon-sur-Mer         0.631096
Toulouse                    0.487665
Charron                     0.487665
Nantes                      0.358577
                              ...   
Bellegarde-sur-Valserine    0.014343
Villeron                    0.014343
SANARY SUR MER              0.014343
Lésigny                     0.014343
NANCY                       0.014343
Name: proportion, Length: 3775, dtype: float64

In [41]:
df["pays"].value_counts(normalize=True) * 100

pays
France                   98.23
Liban                     0.14
Royaume du Maroc          0.08
Turquie                   0.06
République de Tunisie     0.06
                         ...  
Afrique Du Sud            0.01
Tanzanie                  0.01
Pakistan                  0.01
Benin                     0.01
Féd. russe                0.01
Name: proportion, Length: 91, dtype: float64

In [42]:
df["ministere"].value_counts(normalize=True) * 100

ministere
Ecologie - Développement durable - Energie - Aménagement du territoire    25.18
Intérieur - Outre-mer - Collectivités territoriales                       19.41
France Domaine                                                            11.72
Ecologie, développement et aménagement durables.                           7.27
Ecologie, Energie, Dév.dur., Amt.ter.                                      7.03
Intérieur, outre-mer et collectivités territoriales.                       5.51
Intérieur, Outre-mer et Collec. terr.                                      4.97
Justice                                                                    3.50
Budget - Comptes publics - Fonction publique                               2.74
Culture                                                                    1.36
Education nationale - Enseignement supérieur - Recherche                   1.23
Multi-occupants                                                            1.15
Budget, comptes publics et fon

In [43]:
df["date_inventaire"].value_counts(normalize=True) * 100

date_inventaire
2013-05    18.81
2014-03    17.97
2015-10    16.29
2017-03    16.15
2017-12    15.76
2018-12    15.02
Name: proportion, dtype: float64

In [44]:
df[df['code_postal'].isnull()]

,region,ue,id,type,fonction,designation_batiment_terrain,dept,adresse,code_postal,ville,pays,ministere,date_inventaire
0,CORSE,a19e7cbea4c2dae3cfca703bda49e466,936465ba5b332c60491f6cba50a76b49,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,2B,None,None,None,France,"Intérieur, Outre-mer et Collec. terr.",2014-03
1,NORD-PAS DE CALAIS,c1fb47969da8dd1a5a1a93ae33245ced,44b7ec415ee6f810d44fda9c8c585069,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,62,None,None,None,France,Intérieur - Outre-mer - Collectivités territor...,2015-10
70,ILE DE France,7cd5494de9b8558d5b2f21b48d4a0538,14c67822dfeab23b3477e95eb30d0f93,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,92,None,None,None,France,"Intérieur, outre-mer et collectivités territor...",2013-05
71,D.O.M,ba4afce005c7f1434a50c780c7745791,7a9f437aa577b4d83254cb6c262b346a,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,973,None,None,None,France,Intérieur - Outre-mer - Collectivités territor...,2018-12
87,FRANCHE COMTE,0d2de11452a74b1430a29ba9621956ed,e42bfb5e3c7ab0cb177c3c1f0889c079,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,70,None,None,None,France,Intérieur - Outre-mer - Collectivités territor...,2017-12
...,...,...,...,...,...,...,...,...,...,...,...,...,...
9977,D.O.M,9791284abdfd5aa5f63a355718d829ab,ad09e8d218cf2ae59a09d483f845b4bc,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,971,None,None,None,France,Intérieur - Outre-mer - Collectivités territor...,2018-12
9978,CHAMPAGNE-ARDENNES,24da8fc9fed88d94c28567aea4c93c37,d1988c6bf58779a0d084bb28e4c32036,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,10,None,None,None,France,Intérieur - Outre-mer - Collectivités territor...,2015-10
9979,RHONE ALPES,0153d22b410074d22d29b1ee868d2fd6,c77197fd10ced833f749a3f666f0708a,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,74,None,None,None,France,Intérieur - Outre-mer - Collectivités territor...,2015-10
9980,ILE DE France,dc1ca43fd5ef2e8e3d458017c6b02a30,4fe7fdab0058640d9516012ee0b1b43e,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,95,None,None,None,France,"Intérieur, Outre-mer et Collec. terr.",2014-03


In [45]:
df[df['code_postal'].isnull()]['ministere'].value_counts(normalize=True) * 100

ministere
Intérieur - Outre-mer - Collectivités territoriales                       64.378109
Intérieur, outre-mer et collectivités territoriales.                      18.275290
Intérieur, Outre-mer et Collec. terr.                                     16.484245
France Domaine                                                             0.298507
Ecologie - Développement durable - Energie - Aménagement du territoire     0.298507
Affaires étrangères et européennes.                                        0.099502
Services du Premier ministre                                               0.033167
Affaires étrangères                                                        0.033167
Budget - Comptes publics - Fonction publique                               0.033167
Ecologie, développement et aménagement durables.                           0.033167
Budget, comptes publics et fonction publique.                              0.033167
Name: proportion, dtype: float64

In [46]:
df[df['adresse'].isnull()]

,region,ue,id,type,fonction,designation_batiment_terrain,dept,adresse,code_postal,ville,pays,ministere,date_inventaire
0,CORSE,a19e7cbea4c2dae3cfca703bda49e466,936465ba5b332c60491f6cba50a76b49,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,2B,None,None,None,France,"Intérieur, Outre-mer et Collec. terr.",2014-03
1,NORD-PAS DE CALAIS,c1fb47969da8dd1a5a1a93ae33245ced,44b7ec415ee6f810d44fda9c8c585069,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,62,None,None,None,France,Intérieur - Outre-mer - Collectivités territor...,2015-10
14,LIMOUSIN,177659,IG00100075684,Espace naturel,Terrain forestier,terrain nouhant commune,23,None,23170,Nouhant,France,Ecologie - Développement durable - Energie - A...,2017-12
19,PAYS DE LOIRE,174838,IG00100115404,Espace naturel,Terrain en friche,RENFORCEMENT LEVEE VAL D'AUTHION,49,None,49800,Loire-Authion,France,Ecologie - Développement durable - Energie - A...,2018-12
55,AQUITAINE,136699,IG00100039213,Espace naturel,Terrain délaissé,VOIE SUR BERGE DOUBLEMENT RN 113,47,None,47450,Colayrac-Saint-Cirq,France,"Ecologie, Energie, Dév.dur., Amt.ter.",2014-03
...,...,...,...,...,...,...,...,...,...,...,...,...,...
9977,D.O.M,9791284abdfd5aa5f63a355718d829ab,ad09e8d218cf2ae59a09d483f845b4bc,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,971,None,None,None,France,Intérieur - Outre-mer - Collectivités territor...,2018-12
9978,CHAMPAGNE-ARDENNES,24da8fc9fed88d94c28567aea4c93c37,d1988c6bf58779a0d084bb28e4c32036,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,10,None,None,None,France,Intérieur - Outre-mer - Collectivités territor...,2015-10
9979,RHONE ALPES,0153d22b410074d22d29b1ee868d2fd6,c77197fd10ced833f749a3f666f0708a,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,74,None,None,None,France,Intérieur - Outre-mer - Collectivités territor...,2015-10
9980,ILE DE France,dc1ca43fd5ef2e8e3d458017c6b02a30,4fe7fdab0058640d9516012ee0b1b43e,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,95,None,None,None,France,"Intérieur, Outre-mer et Collec. terr.",2014-03


In [47]:
df[df['adresse'].isnull()]['ministere'].value_counts(normalize=True) * 100

ministere
Intérieur - Outre-mer - Collectivités territoriales                       59.032847
Intérieur, outre-mer et collectivités territoriales.                      16.757908
Intérieur, Outre-mer et Collec. terr.                                     15.115572
Ecologie - Développement durable - Energie - Aménagement du territoire     4.957421
France Domaine                                                             1.824818
Ecologie, Energie, Dév.dur., Amt.ter.                                      0.821168
Ecologie, développement et aménagement durables.                           0.760341
Agriculture                                                                0.091241
Services du Premier ministre                                               0.091241
Education nationale - Enseignement supérieur - Recherche                   0.091241
Justice                                                                    0.060827
Culture                                                           

In [48]:
df[df['ville'].isnull()]

,region,ue,id,type,fonction,designation_batiment_terrain,dept,adresse,code_postal,ville,pays,ministere,date_inventaire
0,CORSE,a19e7cbea4c2dae3cfca703bda49e466,936465ba5b332c60491f6cba50a76b49,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,2B,None,None,None,France,"Intérieur, Outre-mer et Collec. terr.",2014-03
1,NORD-PAS DE CALAIS,c1fb47969da8dd1a5a1a93ae33245ced,44b7ec415ee6f810d44fda9c8c585069,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,62,None,None,None,France,Intérieur - Outre-mer - Collectivités territor...,2015-10
70,ILE DE France,7cd5494de9b8558d5b2f21b48d4a0538,14c67822dfeab23b3477e95eb30d0f93,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,92,None,None,None,France,"Intérieur, outre-mer et collectivités territor...",2013-05
71,D.O.M,ba4afce005c7f1434a50c780c7745791,7a9f437aa577b4d83254cb6c262b346a,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,973,None,None,None,France,Intérieur - Outre-mer - Collectivités territor...,2018-12
87,FRANCHE COMTE,0d2de11452a74b1430a29ba9621956ed,e42bfb5e3c7ab0cb177c3c1f0889c079,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,70,None,None,None,France,Intérieur - Outre-mer - Collectivités territor...,2017-12
...,...,...,...,...,...,...,...,...,...,...,...,...,...
9977,D.O.M,9791284abdfd5aa5f63a355718d829ab,ad09e8d218cf2ae59a09d483f845b4bc,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,971,None,None,None,France,Intérieur - Outre-mer - Collectivités territor...,2018-12
9978,CHAMPAGNE-ARDENNES,24da8fc9fed88d94c28567aea4c93c37,d1988c6bf58779a0d084bb28e4c32036,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,10,None,None,None,France,Intérieur - Outre-mer - Collectivités territor...,2015-10
9979,RHONE ALPES,0153d22b410074d22d29b1ee868d2fd6,c77197fd10ced833f749a3f666f0708a,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,74,None,None,None,France,Intérieur - Outre-mer - Collectivités territor...,2015-10
9980,ILE DE France,dc1ca43fd5ef2e8e3d458017c6b02a30,4fe7fdab0058640d9516012ee0b1b43e,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,95,None,None,None,France,"Intérieur, Outre-mer et Collec. terr.",2014-03


In [49]:
df[df['ville'].isnull()]['ministere'].value_counts(normalize=True) * 100

ministere
Intérieur - Outre-mer - Collectivités territoriales                       64.101717
Intérieur, outre-mer et collectivités territoriales.                      18.196830
Intérieur, Outre-mer et Collec. terr.                                     16.413474
France Domaine                                                             0.627477
Ecologie - Développement durable - Energie - Aménagement du territoire     0.429326
Ecologie, développement et aménagement durables.                           0.099075
Justice                                                                    0.033025
Services du Premier ministre                                               0.033025
Affaires étrangères et européennes.                                        0.033025
Agriculture et Pêche                                                       0.033025
Name: proportion, dtype: float64